core/template.yaml

```
@@ -367,11 +411,58 @@ Resources:
                 aws:SourceAccount: !Ref AWS::AccountId
               Bool:
                 aws:SecureTransport: true
+          - Sid: S3LogShipping
+            Effect: Allow
+            Resource:
+              - !Sub ${S3LogsBucket.Arn}
+              - !Sub ${S3LogsBucket.Arn}/*
+            Principal:
+              AWS: '{{resolve:ssm:CSLSS3LambdaARN}}'
+            Action:
+              - s3:List*
+              - s3:Get*
 
   ##########################
   # End: S3 Logs resources #
   ##########################
 
+  ######################
+  # CSLS SQS resources #   
+  ######################
+
+  S3CslsQueue:
+    Type: AWS::SQS::Queue
+    Properties:
+      QueueName: !Sub ${AWS::StackName}-${Environment}-s3-csls-queue
+      KmsMasterKeyId: !GetAtt CslsKmsKey.Arn
+      RedrivePolicy:
+        deadLetterTargetArn: !GetAtt S3CslsDeadLetterQueue.Arn
+        maxReceiveCount: 5
+
+  QueryCompletedQueuePolicy:
+    Type: AWS::SQS::QueuePolicy
+    Properties:
+      Queues:
+        - !Ref S3CslsQueue
+      PolicyDocument:
+        Statement:
+          - Sid: SQSLogshipping
+            Action:
+              - 'sqs:ChangeMessageVisibility'
+              - 'sqs:DeleteMessage'
+              - 'sqs:GetQueueAttributes'
+              - 'sqs:ReceiveMessage'
+            Effect: 'Allow'
+            Resource: !GetAtt S3CslsQueue.Arn
+            Principal:
+              AWS: '{{resolve:ssm:CSLSS3LambdaArn}}'
+
+  S3CslsDeadLetterQueue:
+    Type: AWS::SQS::Queue
+    Properties:
+      QueueName: !Sub ${AWS::StackName}-${Environment}-s3-csls-dead-letter-queue
+      KmsMasterKeyId: !GetAtt CslsKmsKey.Arn
```

Can we make the message retention period 14 days as well? The default is only 4.
```suggestion
      KmsMasterKeyId: !GetAtt CslsKmsKey.Arn
      MessageRetentionPeriod: 1209600
```

---

Done